src/app/components/inputnumber/inputnumber.ts

```diff
@@ -1379,9 +1379,9 @@ export class InputNumber implements OnInit, AfterContentInit, OnChanges, Control
     onInputBlur(event: Event) {
         this.focused = false;
 
-        let newValue = this.validateValue(this.parseValue(this.input.nativeElement.value)).toString();
+        let newValue = this.validateValue(this.parseValue(this.input.nativeElement.value));
         this.input.nativeElement.value = this.formatValue(newValue);
-        this.input.nativeElement.setAttribute('aria-valuenow', newValue);
+        newValue && this.input.nativeElement.setAttribute('aria-valuenow', newValue.toString());
```

Comment:
If the value is `0` (zero), setAttribute won't be executed, although it is a correct value, but falsy.
Also, if the value is `null` or `undefined`, attribute should also be set.